src/devices/cpu/uml.cpp

```diff
@@ -177,9 +177,9 @@ opcode_info const instruction::s_opcode_info_table[OP_MAX] =
 	OPINFO3(SUBB,    "!subb",    4|8, false, C,    SZVC, ALL,  PINFO(OUT, OP, IRM), PINFO(IN, OP, IANY), PINFO(IN, OP, IANY))
 	OPINFO2(CMP,     "!cmp",     4|8, false, NONE, SZVC, ALL,  PINFO(IN, OP, IANY), PINFO(IN, OP, IANY))
 	OPINFO4(MULU,    "!mulu",    4|8, false, NONE, SZV,  ALL,  PINFO(OUT, OP, IRM), PINFO(OUT, OP, IRM), PINFO(IN, OP, IANY), PINFO(IN, OP, IANY)) // Unsigned 32x32=64 and 64x64=128 multiplication
-	OPINFO3(MULUH,   "!muluh",   4|8, false, NONE, SZV,  ALL,  PINFO(OUT, OP, IRM), PINFO(IN, OP, IANY), PINFO(IN, OP, IANY)) // Unsigned 32x32=32 and 64x64=64 multiplication (overflow set based on 32x32=64 calculation but zero and sign based on 32-bit result)
+	OPINFO3(MULULW,  "!mululw",  4|8, false, NONE, SZV,  ALL,  PINFO(OUT, OP, IRM), PINFO(IN, OP, IANY), PINFO(IN, OP, IANY)) // Unsigned 32x32=32 and 64x64=64 multiplication (overflow set based on 32x32=64 calculation but zero and sign based on 32-bit result)
 	OPINFO4(MULS,    "!muls",    4|8, false, NONE, SZV,  ALL,  PINFO(OUT, OP, IRM), PINFO(OUT, OP, IRM), PINFO(IN, OP, IANY), PINFO(IN, OP, IANY)) // Signed 32x32=64 and 64x64=128 multiplication
-	OPINFO3(MULSH,   "!mulsh",   4|8, false, NONE, SZV,  ALL,  PINFO(OUT, OP, IRM), PINFO(IN, OP, IANY), PINFO(IN, OP, IANY)) // Signed 32x32=32 and 64x64=64 multiplication (overflow set based on 32x32=64 calculation but zero and sign based on 32-bit result)
+	OPINFO3(MULSLW,  "!mulslw",  4|8, false, NONE, SZV,  ALL,  PINFO(OUT, OP, IRM), PINFO(IN, OP, IANY), PINFO(IN, OP, IANY)) // Signed 32x32=32 and 64x64=64 multiplication (overflow set based on 32x32=64 calculation but zero and sign based on 32-bit result)
```

Comment:
I’m pretty tired at the moment, but the comment doesn’t make sense: “overflow set based on 32x32=64 calculation”.

It shouldn’t be possible for a 32\*32->64 multiply to overflow.  The largest possible values will always fit in a 64-bit result.

For unsigned:
* 0xffff'ffff \* 0xffff'ffff = 0xffff'fffe'0000'0001

For signed:
* 0x7fff'ffff * 0x7fff'ffff = 0x3fff'ffff'0000'0001
* 0x7fff'ffff * 0x8000'0000 = 0xC000'0000'8000'0000
* 0x8000'0000 * 0x8000'0000 = 0x4000'0000'0000'0000

So is “overflow” representing something other than arithmetic overflow?  Because if it represents arithmetic overflow, it will always be zero, and the results will be the same for the signed and unsigned versions of this instruction (which is why e.g. PowerPC only has `mullw`, since results for the lower half are identical for signed/unsigned, while it has `mulhw` and `mulhwu` because the upper half needs different handling).

Reply:
Going based off the x86 documentation because the flags are calculated the same way:
`If the high-order bits of the product are 0, the CF and OF flags are cleared; otherwise, the flags are set.`

Reply:
x86 flags are pure evil.